test/test_files/function/list.test

```diff
@@ -2353,3 +2353,28 @@ True
 
 -STATEMENT RETURN LIST_CAT(null, null)
 ---- 1
+
+-STATEMENT RETURN LIST_APPEND([], '5')
```

Comment:
We are refactoring our existing test files structures by grouping tests based on the function name.
I would create a new folder under `test/test_files/function` named `string_function`.
Under the `string_function`, we can create two test files:
`list_append.test`: tests for list_append
`list_prepend.test`: tests for list_prepend
And let's move related tests from list.test to `list_append.test` and `list_prepend.test`